src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/crud-table/crud-form/formly-array-type/formly-array-type.component.html

```diff
@@ -0,0 +1,42 @@
+<div class="mb-3">
+  <legend *ngIf="props.label"
+          class="cd-header mt-1"
+          i18n>{{ props.label }}</legend>
+  <p *ngIf="props.description"
+     i18n>{{ props.description }}</p>
+
+  <div *ngFor="let field of field.fieldGroup; let i = index"
+       class="d-flex">
+    <formly-field class="col"
+                  [field]="field"></formly-field>
+    <div class="action-btn">
+      <button class="btn btn-light ms-1"
+              type="button"
+              (click)="addWrapper()">
+        <i [ngClass]="icons.add"></i>
+      </button>
+      <button class="btn btn-light ms-1"
```

Comment:
Given that we're having to introduce all this amount of adaptation/customization, I'm starting to wonder if it'd make sense to just implement our own JSON-based form builder? There's a precedent of this in the Cluster-> MgrModules.

What do you think? Don't take me seriously here, I'm just thinking aloud... :/

Reply:
I'll leave this upto @pereman2 to answer. Personally, I'd rely on external library (one less component for us to manage).

Reply:
and the [formly](https://formly.dev/) looks much more better than the one we were using previously. It got a lot of customization that fits our needs for now atleast.

Reply:
I don't see a reason to develop one ourselves here. The controlSchema is a json schema so we can leave that as it is (there is some extra things I have to remove that don't conform to the json schema), and the uiSchema can be our "special" structure to deal layout/styling stuff.